### PR TITLE
feat: add fields to consumer validators query

### PR DESCRIPTION
**Pull request by sainoe**
## Description

Closes: #2161

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] Added `!` to the type prefix if the change is [state-machine breaking](https://github.com/cosmos/interchain-security/blob/main/RELEASES.md#breaking-changes)
* [ ] Confirmed this PR does not introduce changes requiring state migrations, OR migration code has been added to consumer and/or provider modules
* [ ] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] Provided a link to the relevant issue or specification
* [ ] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/00-intro.md)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)
* [ ] Added a changelog entry to `CHANGELOG.md`
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [ ] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed
* [ ] If this PR is library API breaking, bump the go.mod version string of the repo, and follow through on a new major release

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` the type prefix if the change is state-machine breaking
* [ ] confirmed this PR does not introduce changes requiring state migrations, OR confirmed migration code has been added to consumer and/or provider modules
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

File: proto/interchain_security/ccv/provider/v1/query.proto [hunk 1]
```diff
@@ -342,10 +343,33 @@ message QueryConsumerValidatorsValidator {
...
   // The rate to charge delegators on the consumer chain, as a fraction
-  string rate = 4 [
+  string consumer_commission_rate = 4 [
     (gogoproto.customtype) = "cosmossdk.io/math.LegacyDec",
     (gogoproto.nullable)   = false
     ];
+  // description defines the description terms for the validator
+  cosmos.staking.v1beta1.Description description = 5 [(gogoproto.nullable) = false];
+  // operator_address defines the address of the validator's operator
+  string operator_address = 6 [(cosmos_proto.scalar) = "cosmos.AddressString"];
```
Comment (mpoke) [inline proto/interchain_security/ccv/provider/v1/query.proto:353]: This is the operator address on the provider, right? If so, rename to `provider_operator_address`.

Comment (mpoke) [reply proto/interchain_security/ccv/provider/v1/query.proto:353]: Actually, given that multiple of these fields are part of the Validator struct, maybe just return that, i.e., 
```
cosmos.staking.v1beta1.Validator provider_validator
```

File: x/ccv/provider/keeper/grpc_query.go [hunk 2]
```diff
@@ -346,30 +345,41 @@ func (k Keeper) QueryConsumerValidators(goCtx context.Context, req *types.QueryC
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	for _, v := range consumerValSet {
+	for _, consVal := range consumerValSet {
```
Comment (mpoke) [inline x/ccv/provider/keeper/grpc_query.go:348]: nit: rename `consVal` to `consumerVal` as we usually used cons for consensus (see `consAddr`).

**Review by MSalopek — APPROVED** [review]
Thank you!

Approve!

File: proto/interchain_security/ccv/provider/v1/query.proto (hunk 1 above)
Comment (MSalopek) [reply proto/interchain_security/ccv/provider/v1/query.proto:353]: The code does what was specified in the issue.

Operator address is not relevant for the consumer chain, so it's clear what it refers to.

Comment (MSalopek) [inline proto/interchain_security/ccv/provider/v1/query.proto:353]: This should be:
```suggestion
  string operator_address = 6 [(cosmos_proto.scalar) = "cosmos.ValidatorAddressString"];
```

**Review by MSalopek — CHANGES_REQUESTED** [review]

File: proto/interchain_security/ccv/provider/v1/query.proto [hunk 3]
```diff
@@ -342,10 +343,33 @@ message QueryConsumerValidatorsValidator {
...
+  // jailed defined whether the validator has been jailed from bonded status or not.
+  bool jailed = 7;
+  // status is the validator status (bonded/unbonding/unbonded).
+  cosmos.staking.v1beta1.BondStatus status = 8;
+  // The rate to charge delegators on the provider chain, as a fraction
+  string provider_commission_rate = 9 [
```
Comment (insumity) [inline proto/interchain_security/ccv/provider/v1/query.proto:359]: nit: slightly nicer under `consumer_commission_rate` so both commission rates are next to each other.

File: proto/interchain_security/ccv/provider/v1/query.proto [hunk 4]
```diff
@@ -342,10 +343,33 @@ message QueryConsumerValidatorsValidator {
...
+    (gogoproto.nullable)   = false
+    ];
+  // provider_tokens defines the delegated tokens (incl. self-delegation).
+  string provider_tokens = 10 [
+    (cosmos_proto.scalar)  = "cosmos.Int",
+    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (gogoproto.nullable)   = false
+  ];
+  // The power of the validator used on the provider chain
+  int64 provider_power = 11;
```
Comment (insumity) [inline proto/interchain_security/ccv/provider/v1/query.proto:370]: Shall we deprecate the previously defined `power` and introduce the `consumer_power` field.

**Review by insumity — APPROVED** [review]

File: proto/interchain_security/ccv/provider/v1/query.proto (hunk 4 above)
Comment (MSalopek) [reply proto/interchain_security/ccv/provider/v1/query.proto:370]: That breaks backward compatibility, let's not.

Comment (insumity) [reply proto/interchain_security/ccv/provider/v1/query.proto:370]: How exactly does it break backward compatibility?

Comment (MSalopek) [reply proto/interchain_security/ccv/provider/v1/query.proto:370]: You are renaming a proto field while keeping the same enum index

File: proto/interchain_security/ccv/provider/v1/query.proto (hunk 1 above)
Comment (sainoe) [reply proto/interchain_security/ccv/provider/v1/query.proto]: Renamed to `provider_operator_address`.

Comment (sainoe) [reply proto/interchain_security/ccv/provider/v1/query.proto]: Good catch!

File: proto/interchain_security/ccv/provider/v1/query.proto (hunk 4 above)
Comment (insumity) [reply proto/interchain_security/ccv/provider/v1/query.proto:370]: Please re-read my initial comment:
> Shall we deprecate the previously defined `power` and introduce the `consumer_power` field.

I never talked about renaming anything. I clearly said to deprecate `power` and introduce `consumer_power`. So, something like this:
```
// DEPRECATED blah blah ...
int64 power = 3 [deprecated = true];
...
int64 consumer_power = ...;
```
As far as I know, this is backwards compatible.

Comment (MSalopek) [reply proto/interchain_security/ccv/provider/v1/query.proto:370]: Thanks for pointing it out, my appologies.

I have no preference if that's the way you want to go, as long as there's a field and it's not burried in a nested JSON - great 😄

**Review by MSalopek — APPROVED** [review]

**Pull request merged**